src/transaction/log_page_buffer.c

```diff
@@ -2483,7 +2487,7 @@ logpb_fetch_start_append_page (THREAD_ENTRY * thread_p)
   logpb_log ("started logpb_fetch_start_append_page\n");
 
   /* detect empty log (page and offset of zero) */
-  if ((log_Gl.hdr.append_lsa.pageid == 0) && (log_Gl.hdr.append_lsa.offset == 0))
+  if ((log_Gl.hdr.append_lsa.pageid == log_Gl.hdr.fpageid) && (log_Gl.hdr.append_lsa.offset == 0))
```

Comment:
`log_Gl.hdr.fpageid` is the logical pageid at physical location 1, which seems to be possibly non-zero. For example, in `logpb_next_append_page()`, it is set to the first logical page id of the first one in a new log active volume. If this is comparing logical page id 0, it seems that we should not use `log_Gl.hdr.fpageid`.

Reply:
@hornetmj It seems like the simplest way. It is a hidden and debug-only parameter so that looks good to me.